analyzers/tests/SonarAnalyzer.UnitTest/Rules/SymbolicExecution/LocksShouldBeReleasedTest.cs

```diff
@@ -0,0 +1,58 @@
+﻿/*
+ * SonarAnalyzer for .NET
+ * Copyright (C) 2015-2021 SonarSource SA
+ * mailto: contact AT sonarsource DOT com
+ *
+ * This program is free software; you can redistribute it and/or
+ * modify it under the terms of the GNU Lesser General Public
+ * License as published by the Free Software Foundation; either
+ * version 3 of the License, or (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+ * Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public License
+ * along with this program; if not, write to the Free Software Foundation,
+ * Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301, USA.
+ */
+
+using System.Collections.Generic;
+using System.Linq;
+using Microsoft.CodeAnalysis;
+using Microsoft.VisualStudio.TestTools.UnitTesting;
+using SonarAnalyzer.Rules.CSharp;
+using SonarAnalyzer.UnitTest.MetadataReferences;
+using SonarAnalyzer.UnitTest.TestFramework;
+
+namespace SonarAnalyzer.UnitTest.Rules
+{
+    [TestClass]
+    public class LocksShouldBeReleasedTest
+    {
+        [TestMethod]
+        [TestCategory("Rule")]
+        public void LocksShouldBeReleased_Monitor_CS() =>
+            Verifier.VerifyAnalyzer(@"TestCases\SymbolicExecution\Sonar\LocksShouldBeReleased.Monitor.cs", new LocksShouldBeReleased());
+
+        [TestMethod]
+        [TestCategory("Rule")]
+        public void LocksShouldBeReleased_SpinLock_CS() =>
+            Verifier.VerifyAnalyzer(@"TestCases\SymbolicExecution\Sonar\LocksShouldBeReleased.SpinLock.cs", new LocksShouldBeReleased());
+
+        [TestMethod]
+        [TestCategory("Rule")]
+        public void LocksShouldBeReleased_ReaderWriterLock_CS() =>
+            Verifier.VerifyAnalyzer(@"TestCases\SymbolicExecution\Sonar\LocksShouldBeReleased.ReaderWriterLock.cs", new LocksShouldBeReleased(), GetReferences());
+
+        [TestMethod]
+        [TestCategory("Rule")]
+        public void LocksShouldBeReleased_ReaderWriterLockSlim_CS() =>
+            Verifier.VerifyAnalyzer(@"TestCases\SymbolicExecution\Sonar\LocksShouldBeReleased.ReaderWriterLockSlim.cs", new LocksShouldBeReleased());
+
+        private static IEnumerable<MetadataReference> GetReferences() =>
+            Enumerable.Empty<MetadataReference>()
```

Comment:
What's this for? Once removed, the `MetadataReferenceFacade.SystemThreading` can be directly in the `LocksShouldBeReleased_ReaderWriterLock_CS` call.
```suggestion
```